### PR TITLE
Let admins delete demo accounts

### PR DESCRIPTION
**Pull request by cjdenio**
This PR adds a button that lets admins clean up demo accounts. It's all `acts_as_paranoid`-ified.

I'm thinking users might be able to delete their demo accounts in the future as well.

Screenshot 2023-01-30 at 12 46 14 PM

### CODE REVIEW

File: app/policies/event_policy.rb [hunk 1]
```diff
@@ -53,7 +53,7 @@ def update?
   end
 
   def destroy?
-    user&.admin?
+    user&.admin? && record.demo_mode?
```
Comment (maxwofford) [inline app/policies/event_policy.rb]: Was there an admin destroy route that used to exist for regular orgs?

**Review by maxwofford — COMMENTED** [review]
Thanks! I'm sure the ops team will be happy to have this.

How does this handle things like https://github.com/hackclub/bank/blob/b7f7e5f64e9bb23052f69404a51ffe5105798d6e/app/services/user_service/can_open_demo_mode.rb#L18?

File: app/policies/event_policy.rb (hunk 1 above)
Comment (cjdenio) [reply app/policies/event_policy.rb]: Yeah, there already was one (though it's currently broken and I can't find it in the UI)

**Comment by cjdenio** [comment]
Since the event is effectively deleted, another demo account slot will be freed up when a demo account is deleted.

**Review by garyhtou — APPROVED** [review]
LGTM. just to double check, it's not possible for demo mode events to have financial data, right? (no disbursements or card charges, etc.)

**Comment by cjdenio** [comment]
Yep! Demo mode accounts don't have any associated financial data

**Comment by cjdenio** [comment]
Reviewed by @maxwofford IRL.

**Pull request merged**